src/module.ts

```diff
@@ -38,9 +38,20 @@ export class SsvUxModule {
 export function _moduleConfigFactory(config: UxOptions | (() => UxOptions)): UxOptions {
 	const uxOptions = (typeof config === "function" ? config() : config) || UX_DEFAULT_CONFIG;
 	const viewport = {
-		...UX_DEFAULT_CONFIG.viewport,
-		...uxOptions.viewport
-	}; // breakpoints shouldn't be merged
+		resizePollingSpeed: uxOptions?.viewport?.resizePollingSpeed
+			? uxOptions.viewport.resizePollingSpeed
+			: UX_DEFAULT_CONFIG.viewport.resizePollingSpeed,
+		breakpoints: uxOptions?.viewport?.breakpoints
+			? uxOptions.viewport.breakpoints
+			: UX_DEFAULT_CONFIG.viewport.breakpoints,
+
+		viewportSsrSizes: {
+			...UX_DEFAULT_CONFIG.viewport.viewportSsrSizes,
+			...(uxOptions?.viewport?.viewportSsrSizes
+				? uxOptions.viewport.viewportSsrSizes
+				: {}),
+		},
+	};
```

Comment:
not entirely sure what you're trying to achieve, but it was much cleaner as it was